docs/config/README.md

```diff
@@ -3162,6 +3163,21 @@ By default the module will be shown if any of the following conditions are met:
 format = 'via [🔹 $version](147 bold) '
 ```
 
+## Pijul Channel
+
+The `pijul` module shows the active channel of the repo in your current directory.
+
+### Options
+
+| Option              | Default                           | Description                                                                          |
+| ------------------- | --------------------------------- | ------------------------------------------------------------------------------------ |
+| `symbol`            | `' '`                            | The symbol used before the pijul channel name of the repo in your current directory. |
+| `style`             | `'bold purple'`                   | The style for the module.                                                            |
+| `format`            | `'on [$symbol$channel]($style) '` | The format for the module.                                                           |
+| `truncation_length` | `2^63 - 1`                        | Truncates the pijul channel name to `N` graphemes                                    |
```

Comment:
To be more compatible with 32-bit systems:
```suggestion
| `truncation_length` | `2^32 - 1`                        | Truncates the pijul channel name to `N` graphemes                                    |
```

Reply:
Both `git_branch` and `hg_branch` take `i64` for `truncation_length`. While I agree that this is an easy change for 32-bit compatibility, I think consistency is more important. All of the modules that use truncation need to be using `usize` instead of `i64`.

Reply:
Either way, I feel like an unsigned integer type should be preferred and meson also uses `u32`.

Reply:
Sure, but let's do a follow-up change to handle all of the affected modules, instead of just polishing this one.

Reply:
I've avoided doing this, because it could be a breaking change. If for instance they are switched to `u32` the former default would be outside the accepted value range, and negative values would lead to errors instead of warnings.

Reply:
At the risk of causing [XKCD 1172](https://xkcd.com/1172/), I doubt anything would _actually_ break. Defaults shouldn't be stored in any TOML files, and anyone who actually sets `truncation_length` for any module is most likely to set it less than 200. As for handling negative values, I wonder if there's a way to adjust the deserializer to emit warnings with defaults instead of hard errors. Possibly we could make the default `0` instead of `INT_MAX`, as at these sizes, there's no practical difference.

Reply:
I think some people will have based their config-file on `starship print-config --default`, it was even recommended in the docs for a few weeks.

Reply:
Oof. That does make a mess of things then.